### PR TITLE
qa/tasks/cephadm: learn to pull cephadm from github

### PR DESCRIPTION
**Pull request by liewegas**
fixes https://tracker.ceph.com/issues/43755

### CODE REVIEW

**Comment by jan--f** [comment]
The cephadm download now works for me, but something else in the cephadm task fails

http://pulpito.front.sepia.ceph.com/jafaj-2020-01-23_08:26:56-jantest-master-distro-basic-smithi/4696329/

**Comment by sebastian-philipp** [comment]
> The cephadm download now works for me, but something else in the cephadm task fails
> 
> http://pulpito.front.sepia.ceph.com/jafaj-2020-01-23_08:26:56-jantest-master-distro-basic-smithi/4696329/

```
2020-01-23T08:43:16.079 INFO:tasks.cephadm:Bootstrapping...
2020-01-23T08:43:16.080 INFO:teuthology.orchestra.run.smithi068:> sudo /home/ubuntu/cephtest/cephadm --image quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497 bootstrap --fsid 65a75ff8-3dbc-11ea-99dd-001a4aab830c --mon-id a --mgr-id y --config /home/ubuntu/cephtest/seed.ceph.conf --output-config /home/ubuntu/cephtest/ceph.conf --output-keyring /home/ubuntu/cephtest/ceph.keyring --output-pub-ssh-key /home/ubuntu/cephtest/ceph.pub --mon-ip 172.21.15.68 && sudo chmod +r /home/ubuntu/cephtest/ceph.keyring
2020-01-23T08:43:16.135 INFO:teuthology.orchestra.run.smithi068.stderr:/home/ubuntu/cephtest/cephadm: 1: /home/ubuntu/cephtest/cephadm: 404:: not found
2020-01-23T08:43:16.137 DEBUG:teuthology.orchestra.run:got remote process result: 127
```

Looks suspiciously as if something went wrong with the download.

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -88,24 +88,42 @@ def download_cephadm(ctx, config, ref):
...
-                'test', '-s',
-                ctx.cephadm,
-                run.Raw('&&'),
-                'chmod', '+x',
-                ctx.cephadm,
-            ],
-        )
+        if git_url.startswith('https://github.com/'):
+            # git archive doesn't like https:// URLs, which we use with github.
+            rest = git_url.split('https://github.com/', 1)[1]
```
Comment (jan--f) [inline qa/tasks/cephadm.py]:
```suggestion
            rest = git_url.split('https://github.com/', 1)[1].split('.git')[0]
```

We need to loose any '`.git` prefix it its there.

**Comment by jan--f** [comment]
Hmm with this I now get
`2020-01-23T10:00:56.342 INFO:teuthology.orchestra.run.smithi027.stderr:INFO:cephadm:/usr/bin/docker:stderr Error response from daemon: manifest for quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497 not found`

Anything I'm doing wrong in my invocation?
`teuthology-suite -k distro --priority 50 --machine-type smithi --suite jantest --email jfajeski@suse.com --ceph-repo https://github.com/ceph/ceph --ceph master --newest 20 --suite-repo https://github.com/jan--f/ceph.git --suite-branch teuth-test`

The jantest suite is just a stripped down cephadm deployment to log in: https://github.com/jan--f/ceph/commit/720a49fae0123ed193c8c79ffdc8b50bb6243686

**Comment by sebastian-philipp** [comment]
> Hmm with this I now get
> `2020-01-23T10:00:56.342 INFO:teuthology.orchestra.run.smithi027.stderr:INFO:cephadm:/usr/bin/docker:stderr Error response from daemon: manifest for quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497 not found`

Indeed. I'm getting 

```
➜  ceph2 git:(master) podman pull quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497
Trying to pull quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497...
  manifest unknown: manifest unknown
Error: error pulling image "quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497": unable to pull quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497: unable to pull image: Error initializing source docker://quay.io/ceph-ci/ceph:f205d38c2c1940a4247ebdb4e795d18266d94497: Error reading manifest f205d38c2c1940a4247ebdb4e795d18266d94497 in quay.io/ceph-ci/ceph: manifest unknown: manifest unknown
```
Looks like as if the image wasn't uploaded to quay.
You have to make sure the images are uploaded to quay. I don't exactly know how that is done and works with non-standard git repository urls. (other than `ceph/ceph-ci` and `ceph/ceph`)

> 
> Anything I'm doing wrong in my invocation?
> `teuthology-suite -k distro --priority 50 --machine-type smithi --suite jantest --email jfajeski@suse.com --ceph-repo https://github.com/ceph/ceph --ceph master --newest 20 --suite-repo https://github.com/jan--f/ceph.git --suite-branch teuth-test`

priority 50 is a bit low imo. See https://docs.ceph.com/docs/master/dev/developer_guide/tests-integration-tests/#testing-priority

> 
> The jantest suite is just a stripped down cephadm deployment to log in: [jan--f@720a49f](https://github.com/jan--f/ceph/commit/720a49fae0123ed193c8c79ffdc8b50bb6243686)

**Comment by jan--f** [comment]
> Looks like as if the image wasn't uploaded to quay.
> You have to make sure the images are uploaded to quay. I don't exactly know how that is done and works with non-standard git repository urls. (other than `ceph/ceph-ci` and `ceph/ceph`)

I am using `--ceph-repo https://github.com/ceph/ceph --ceph master` though. Only the suite repo is my own.

File: qa/tasks/cephadm.py (hunk 1 above)
Comment (liewegas) [reply qa/tasks/cephadm.py]: added a patch to wrok around this.  in general, you should just leave the .git off when scheduling the suite (e.g., ``teuthology-suite --suite-repo https://github.com/liewegas/ceph ...``)

**Pull request merged**